src/cell.c

```diff
@@ -61,15 +61,14 @@ static int *get_overlap_table(const VecDBL *position, int cell_size,
 // @param tensor_rank rank of site tensors for magnetic symmetry. Set -1 if not
 // used.
 Cell *cel_alloc_cell(const int size, const SiteTensorType tensor_rank) {
-    Cell *cell;
-
-    cell = NULL;
-
+    // NOLINTBEGIN(bugprone-easily-swappable-parameters)
     if (size < 1) {
         return NULL;
     }
 
-    if ((cell = (Cell *)malloc(sizeof(Cell))) == NULL) {
+    Cell *cell = (Cell *)malloc(sizeof(Cell));
+
+    if (cell == NULL) {
```

Comment:
Here's an example of
- avoiding double instantiation
- report from `clang-tidy` discourages usage of variable instantiation in  the if loop. If an error occurs in the function call, the return value can be ill-defined